client/src/components/graph/graph.tsx

```diff
@@ -1033,68 +1054,98 @@ class Graph extends React.Component<GraphProps, GraphState> {
       const offscreenCanvas = document.createElement("canvas");
       offscreenCanvas.width = width;
       offscreenCanvas.height = height;
-      const legendNodes =
-        document.getElementById("continuous_legend")?.childNodes;
+
       const ctx = offscreenCanvas.getContext("2d");
       ctx?.drawImage(graph, 0, 0);
 
-      legendNodes?.forEach(async (node) => {
-        if (node.nodeName === "CANVAS") {
-          // get style from node
-          const style = window.getComputedStyle(node as HTMLCanvasElement);
-          const nodeWidth = parseInt(style.width, 10);
-          const nodeHeight = parseInt(style.height, 10);
-          const nodeLeft = parseInt(style.left, 10);
-          const nodeTop = parseInt(style.top, 10);
-          ctx?.drawImage(
-            node as HTMLCanvasElement,
-            PADDING + nodeLeft,
-            PADDING + nodeTop,
-            nodeWidth,
-            nodeHeight
+      let categoricalLegendImageURI: string | null = null;
+
+      if (colors.colorMode !== null) {
+        if (colors.colorMode !== "color by categorical metadata") {
+          const legendNodes =
+            document.getElementById("continuous_legend")?.childNodes;
+          legendNodes?.forEach(async (node) => {
+            if (node.nodeName === "CANVAS") {
+              // get style from node
+              const style = window.getComputedStyle(node as HTMLCanvasElement);
+              const nodeWidth = parseInt(style.width, 10);
+              const nodeHeight = parseInt(style.height, 10);
+              const nodeLeft = parseInt(style.left, 10);
+              const nodeTop = parseInt(style.top, 10);
+              ctx?.drawImage(
+                node as HTMLCanvasElement,
+                PADDING + nodeLeft,
+                PADDING + nodeTop,
+                nodeWidth,
+                nodeHeight
+              );
+            } else if (node.nodeName === "svg" || node.nodeName === "SVG") {
+              const svgNode = node as SVGSVGElement;
+              const xml = new XMLSerializer().serializeToString(svgNode);
+              const svg64 = btoa(unescape(encodeURIComponent(xml)));
```

Comment:
I'm getting deprecated usage warning for `unescape` - worth checking 🙆‍♂️ ?

<img width="945" alt="Screenshot 2024-04-15 at 8 02 52 AM" src="https://github.com/chanzuckerberg/single-cell-explorer/assets/6309723/e8e7821a-b0b5-479c-ad63-6412a93c6447">